iBurnApp/iBurn-iOS
# Update `BRCEventObjectTableViewCell` to Swift and MVVM

Update `BRCEventObjectTableViewCell` to Swift and MVVM
* Introduces protocols to aid in testing

Tested:
* Type "Event" in Search
* Ensure UI matches pre-PR
* Unit tested all work

BEFORE
<img width="565" alt="Event List - BEFORE" src="https://github.com/iBurnApp/iBurn-iOS/assets/1483008/ceeb1f6e-44dd-4d2b-9e2f-6d7983dda023">


AFTER

<img width="565" alt="Event List - AFTER" src="https://github.com/iBurnApp/iBurn-iOS/assets/1483008/fe1e54e1-4c49-4758-8c72-0dfef333b4a5">
